### ISSUE TITLE
Cannot install Plex

### ISSUE THREAD

**Issue by NickJorens** [rockstor/rockstor-core#1466]
I installed plex before but deleted it to change sometings,
now when I try to reinstall, i get this:
- I click on install it opens the menu i fill in everything, then click ok, the window closes and goes to the installed rock-ons tab. But i never show. This is the log

[07/Oct/2016 01:07:42] DEBUG [storageadmin.util:48] Current Rockstor version: 3.8-14.17
[07/Oct/2016 01:11:15] DEBUG [storageadmin.views.command:152] Bootstrap operations completed
[07/Oct/2016 01:11:36] DEBUG [storageadmin.views.rockon_helpers:70] Attempted to remove a container(plex-linuxserver.io). out: [''] err: ['Error response from daemon: no such id: plex-linuxserver.io', 'Error: failed to remove containers: [plex-linuxserver.io]', ''] rc: 1.
[07/Oct/2016 01:11:36] DEBUG [storageadmin.views.rockon_helpers:127] exception while installing the Rockon(22)
[07/Oct/2016 01:11:36] ERROR [storageadmin.views.rockon_helpers:128] Error running a command. cmd = ['/usr/bin/docker', 'run', '-d', '--restart=unless-stopped', '--name', 'plex-linuxserver.io', '-v', '/mnt2/plex-config:/config', '-v', '/mnt2/plex-data:/data', '-v', '/etc/localtime:/etc/localtime:ro', '-p', '32400:32400/tcp', '-p', '32400:32400/udp', '--net=host', '-e', 'PUID=nick', '-e', 'VERSION=latest', '-e', 'PGID=root', 'linuxserver/plex']. rc = 1. stdout = ['']. stderr = ['Error response from daemon: could not find image: no such id: 4b7012161965f32a83f60d3fd724bcfc84d6a01c098b12c004f82999cd96eb4c', '']
Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/storageadmin/views/rockon_helpers.py", line 125, in install
    globals().get('%s_install' % rockon.name.lower(), generic_install)(rockon)
  File "/opt/rockstor/src/rockstor/storageadmin/views/rockon_helpers.py", line 216, in generic_install
    run_command(cmd)
  File "/opt/rockstor/src/rockstor/system/osi.py", line 104, in run_command
    raise CommandException(cmd, out, err, rc)
CommandException: Error running a command. cmd = ['/usr/bin/docker', 'run', '-d', '--restart=unless-stopped', '--name', 'plex-linuxserver.io', '-v', '/mnt2/plex-config:/config', '-v', '/mnt2/plex-data:/data', '-v', '/etc/localtime:/etc/localtime:ro', '-p', '32400:32400/tcp', '-p', '32400:32400/udp', '--net=host', '-e', 'PUID=nick', '-e', 'VERSION=latest', '-e', 'PGID=root', 'linuxserver/plex']. rc = 1. stdout = ['']. stderr = ['Error response from daemon: could not find image: no such id: 4b7012161965f32a83f60d3fd724bcfc84d6a01c098b12c004f82999cd96eb4c', '']

**Comment by phillxnet** [comment]
@NickJorens @schakrava There have been a few changes to rock-ons management code but most notable that comes to mind is:
is pr: #1795 "explicitly pull an image to get potential updates."
for issue #1795 "rock-on UI fails to reflect removed container"
I suspect the above code improvements would help in this case.
@NickJorens Is this issue still reproducible with the above fix in place?

**Comment by CJandDarren** [comment]
Well, I have just gotten Rockstor 3.9.2-53 up and running but when I go to install plex it does the installing wheel then just disappears. I looked in the /opt/rockstor/var/log but did not find the right log I think. I went through a few of them but found nothing with Plex in it. 

If you need any information or testing just yell.

**Comment by FroggyFlox** [comment]
Hi @CJandDarren ,

I see this kind of behavior some times when running in a VM that has some networking issues and thus cannot contact docker hub to pull the image. I wonder if you suffer from something similar. 

I thus would have a very stupid suggestion but I still need to: if you haven't done so already, I would first make sure the issue is still there after a complete shutdown and reboot of the machine so that everything is reset. 

Otherwise, I would recommend a few sanity checks:
`docker info`
`docker ps -a`
And then manually trying to pull the image to make sure your machine can reach docker.io correctly:
`docker pull linuxserver/plex`

If the latter is successful, I would try installing the rockon again and see if it works. We do have some improvement to make in order to better resurface this kind of silent failures but that should come in the future.

Hope this helps,

**Comment by CJandDarren** [comment]
> Hi @CJandDarren ,
> 
> I see this kind of behavior some times when running in a VM that has some networking issues and thus cannot contact docker hub to pull the image. I wonder if you suffer from something similar.
> 
> I thus would have a very stupid suggestion but I still need to: if you haven't done so already, I would first make sure the issue is still there after a complete shutdown and reboot of the machine so that everything is reset.
> 
> Otherwise, I would recommend a few sanity checks:
> `docker info`
> `docker ps -a`
> And then manually trying to pull the image to make sure your machine can reach docker.io correctly:
> `docker pull linuxserver/plex`
> 
> If the latter is successful, I would try installing the rockon again and see if it works. We do have some improvement to make in order to better resurface this kind of silent failures but that should come in the future.
> 
> Hope this helps,

Here is what I got, from the suggested commands. 


**` docker info`**
```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 17.09.0-ce
Storage Driver: btrfs
 Build Version: Btrfs v4.9.1
 Library Version: 102
Logging Driver: journald
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.12.4-1.el7.elrepo.x86_64
Operating System: Rockstor 3 (Core)
OSType: linux
Architecture: x86_64
CPUs: 12
Total Memory: 62.92GiB
Name: lankhmar
ID: VIW5:O6UH:23IO:LCAP:PTGB:LHQ4:P367:ZVRL:TTPB:BRS3:CPZ6:WOAD
Docker Root Dir: /mnt2/Rock_ons
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

     ```
docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES

```

docker pull linuxserver/plex
Using default tag: latest
latest: Pulling from linuxserver/plex
Digest: sha256:fd3e75fca88e7e6d253e7c5e2825c4d0097cce241268148bcd6d6bfaa4f54818
Status: Image is up to date for linuxserver/plex:latest

**Comment by CJandDarren** [comment]
Tried reinstalling after a reboot. Nada still not installing.

**Comment by FroggyFlox** [comment]
> Tried reinstalling after a reboot. Nada still not installing.

Thanks for the additional information. I just saw your posts on the forum and it seems the cause of this current issue relates to what you described in the forum. We'll thus need to figure that one out first.

**Comment by CJandDarren** [comment]
Sorted out that issue. it was me being forgetful and not adding the rockstor directory. 
but still no luck with installin it after fixing that.

**Comment by phillxnet** [comment]
@CJandDarren Hello again.
Could you make certain your Rockstor verrsion via:
```
yum info rockstor
```
and ensure that you have done all the steps as descried in our doc on installing Plex:

http://rockstor.com/docs/docker-based-rock-ons/plex-media-server.html

Please also note that we would be best to pin this down prior to contributing to this now 4 year old issue. It's simply not the case that our Plex Rock-on hasn't worked for 4 years. I should have closed it yes and that is what I will do when we are done here as you and @FroggyFlox have now got some info in place but as stated in my response to the original poster we have much newer code for this stuff and there has been no other reports of Plex issues to follow up on the original here. We should probably have some kind of auto close thing going on with our issue I think. If we find your particular issue exactly we should open a fresh one for that I think. So lest pick this up in the forum as our Plex rock-on has worked for a number of years since this issue and this issue should have been closed long ago. Apologies for leaving it open.

The forum has more eyes and issue need exact reproducers ideally so that developers can reproduce and thus see if what they propose as a fix actually is a fix. This issue never got one, hence it's my slip up for not closing it years ago!

So lest pick this up in the forum and then create a new clean issue with a reproducer for your issue. I often use the Plex rock-on as a test case and did so only the other day so it is likely we have some kind of doc issue where your setup is not what it should be and if it's actually a new issue then it should be reported as such, which bring this round to my negligence in going back over the really old issue. Oh well.

Thanks for contributing so generously though and don't be put off, I'm just conscious of this looking like our Plex hasn't worked for 4 years !!! Obviously not the case. 

Another point to developing reproducers and finding the issue first in the forum is that we have better facilities and the 'process' of finding the 'problem' can be shared with more users. GitHub is best left for the actual reproducer so the move to a fix can begin. That way other users having difficulty getting a Rock-on setup working can learn from the discussion within the forum. It also keeps the GitHub issues smaller and easier for developers to read their entire history.

I hope that makes sense and I know it's good to contribute to existing issues, hence this being my fault for not closing this very old issue. Unless of course your issue is exactly the same which is highly unlikely given 4 years of updates and us using much changed Rock-on code and an entirely updated plex docker image to boot in the mean time.

So in summary we try to develop issues to the point of some initial understanding and ideally a reproducer before we bring them to GitHub. But it's an open system so this is often not understood by our new users. So maybe what we need here is some improved docs. If it's not one thing it's another ;). Thanks again for contributing by the way.

**Comment by phillxnet** [comment]
@FroggyFlox  and @CJandDarren 
This issue is earmarked for closure if @CJandDarren current report turns out to not be related to the original poster. Apologies all, I need to do some tidy up.
